### ISSUE TITLE
App routinely routes me to walk through the traffic in the middle of an intersection

### ISSUE THREAD

**Issue by kuba-orlik** [organicmaps/organicmaps#2816]
I'm not sure if it's an issue with map data, or something changed in the routing engine, but every other day the app struggles with intersections when planning a pedestrian route. More often than not instead of routing through pedestrian paths, it just wants me to walk halfway through the pedestrian crossing, join car traffic and walk through the middle of the intersection/roundabout:

![image](https://user-images.githubusercontent.com/2697916/175489816-4f483621-9d54-4742-9a49-772a5405bb46.png)
![image](https://user-images.githubusercontent.com/2697916/175489841-80c47798-2f38-4323-993d-c820120a3ef5.png)
![image](https://user-images.githubusercontent.com/2697916/175489846-a0497031-c3c4-4bbb-998c-b818210f97fd.png)

The screenshots relate to these two locations on the map, but the issue is not unique to those:

* https://www.openstreetmap.org/search?query=pozna%C5%84#map=19/52.39695/16.89905
* https://www.openstreetmap.org/search?query=rondo%20obornickie#map=17/52.44333/16.91066

**Comment by vng** [comment]
Yes, this routes looks confusing and the algo should be enhanced.

Here are some tips:
- Actually, we can't remove general highway-primary/secondary roads from pedestrian routing, because of different map coverage and mapping rules. It concerns the first problem. Now somebody already put foot=no for the [roundabout](https://www.openstreetmap.org/way/282420901) so the route should be better after the next maps update.
- If you are building pedestrian route, try to set start/finish point closer to the pedestrian footways. It should fix the second problem.
